### PR TITLE
Feat/update metadata per field

### PR DESCRIPTION
**Pull request by sunguru98**
Created seperate update subcommands to support updating just the `name`, `symbol`, `creators` fields

### CODE REVIEW

**Comment by sunguru98** [comment]
The API and the respective test Tx are as follows

1. Name
Screenshot 2022-02-14 at 4 52 59 AM
TX - https://solscan.io/tx/5eL2aq3DFMuseuhoAHhhXQhva1wpT86x6iwDHqMBsoSXzgpyJux1LVpje7N7DUoCXxhzznYtbmV7F69oWmJMcyUp?cluster=devnet

2. Symbol 
Screenshot 2022-02-14 at 4 54 37 AM
TX - https://solscan.io/tx/24mHU9ywHeFPuXB3J7J6y3XfCr59DEFvWBjCXK1q67jHiaKpX3cegjZPzTUrjQFhAnCAMFMSWYX53hxSLTFXA6rs?cluster=devnet

3. Creators
Screenshot 2022-02-14 at 4 55 17 AM
TX (With custom position of 1) - https://solscan.io/tx/4v3LyzLzntoDxnjJQsWPvwRa6ebAoZkk8dhWwNXMk4xKY3PAnzFbA4MJJxfRdPtR5hHwy1mDMcViVPhvW4N8qm3P?cluster=devnet

TX (Without custom position) for index 0 - https://solscan.io/tx/3fBe3o4dmHJqiPZBcSdq8U6iGzyztFLm8bia7TYGPzTVy7HQ8GcyZMZqcsvbZf9r6oMnpHSxwQyGmbDcBT2qRUUx?cluster=devnet

**Comment by samuelvanderwaal** [comment]
Awesome, thanks! I will review this tonight.

**Comment by sunguru98** [comment]
Aye aye :)

File: src/update_metadata.rs [hunk 1]
```diff
@@ -27,6 +27,88 @@ use crate::decode::{decode, get_metadata_pda};
...
+    let is_verified = parsed_keypair.pubkey().eq(&new_creator_pb);
+    let new_creator = Creator {
+        address: new_creator_pb,
+        share: *new_share,
+        verified: is_verified,
+    };
+
+    let new_creators = match data_with_old_creators.creators {
+        Some(mut old_creators) => {
+            let _creator = std::mem::replace(&mut old_creators[*position as usize], new_creator);
```
Comment (samuelvanderwaal) [inline src/update_metadata.rs:94]: `std::mem::replace` is a clever solution for swapping out creators that already exist. I think we want the API to also allow adding creators to positions that don't exist yet though. If there's a creator's array with only one creator (at position 0) I think it makes sense to allow someone to add an additional creator at position 1. What do you think?

**Review by samuelvanderwaal — CHANGES_REQUESTED** [review]

**Comment by samuelvanderwaal** [comment]
This is great work! There are just a couple of things to address and then it should be good to merge.

**Comment by samuelvanderwaal** [comment]
Take a look at the changes I made and let me know if you have any questions. 

* Removed references on numbers, as primitives like `usize` are copied freely on the stack with no penalty
* Changed `u64` to `usize` to avoid the frequent casting
* Fixed off-by-one issues with the creators array length and the position
* Push creator to end of array instead of trying to index into a non-existing position

Once you're happy with the changes I'll merge.

**Comment by sunguru98** [comment]
This is perfect :) I thought we had like a mandatory requirement to just use the primitive u8 and u64 haha. Im more than happy with these filters

**Pull request merged**